Assets/MRTK/SDK/Experimental/Features/UX/BoundsControl/Visuals/TranslationHandles.cs

```diff
@@ -0,0 +1,338 @@
+                }
+
+                // create new visual
+                Bounds visualBounds = CreateVisual(i, handles[i].gameObject);
+
+                // update handle collider bounds
+                UpdateColliderBounds(handles[i], visualBounds.size);
+            }
+
+            objectsChangedEvent.Invoke(this);
+        }
+
+        protected override void UpdateColliderBounds(Transform handle, Vector3 visualSize)
+        {
+            Debug.Log("UpdateColliderBounds");
+            var invScale = visualSize.x == 0.0f ? 0.0f : config.HandleSize / visualSize.x;
+            GetVisual(handle).transform.localScale = new Vector3(invScale, invScale, invScale);
+            Vector3 colliderSizeScaled = visualSize * invScale;
+            if (config.TranslationHandlePrefabColliderType == HandlePrefabCollider.Box)
+            {
+                BoxCollider collider = handle.gameObject.GetComponent<BoxCollider>();
+                collider.size = colliderSizeScaled;
+                collider.size += BaseConfig.ColliderPadding;
+            }
+            else
+            {
+                SphereCollider collider = handle.gameObject.GetComponent<SphereCollider>();
+                collider.radius = colliderSizeScaled.x * 0.5f;
+                collider.radius += VisualUtils.GetMaxComponent(config.ColliderPadding);
+            }
+        }
+
+
+        protected Bounds CreateVisual(int handleIndex, GameObject parent)
+        {
+            GameObject midpointVisual;
+            GameObject prefabType = config.HandlePrefab;
+            if (prefabType != null)
+            {
+                midpointVisual = Object.Instantiate(prefabType);
+            }
+            else
+            {
+                midpointVisual = GameObject.CreatePrimitive(PrimitiveType.Sphere);
+                Object.Destroy(midpointVisual.GetComponent<SphereCollider>());
+            }
+
+            // Even handle indices point in the positive direction along their axis,
+            // and odd handle indices point in the negative direction;
+            var directionSign = handleIndex % 2 == 0 ? 1.0f : -1.0f;
+
+            // Align handle with its edge assuming that the prefab is initially aligned with the up direction 
+            if (faceAxes[handleIndex] == CardinalAxisType.X)
+            {
+                Quaternion realignment = Quaternion.FromToRotation(Vector3.forward, directionSign * Vector3.right);
+                midpointVisual.transform.localRotation = realignment * midpointVisual.transform.localRotation;
+            }
+            else if (faceAxes[handleIndex] == CardinalAxisType.Z)
+            {
+                Quaternion realignment = Quaternion.FromToRotation(Vector3.forward, directionSign * Vector3.forward);
+                midpointVisual.transform.localRotation = realignment * midpointVisual.transform.localRotation;
+            } else if (faceAxes[handleIndex] == CardinalAxisType.Y)
```

Comment:
Is it possible for none of these if/else statements to be true?

Reply:
@thalbern ?

Reply:
there's only 3 types of axis so there shouldn't be a scenario that we don't catch - however I changed the code in translation handles now that the Y case is the fallback (else) in case someone adds another axis / value to that enum